setup.py

```diff
@@ -1,75 +1,45 @@
-    return locals()['version']
-
-
-setup(
-    name='eralchemy2',
-
-    version=read_version(),
-
-    description='Simple entity relation (ER) diagrams generation',
-    long_description=long_description,
-
-    # The project's main homepage.d
-    url='https://github.com/maurerle/eralchemy2',
-
-    # Author details
-    author='Florian Maurer',
-    author_email='fmaurer+github@disroot.org',
-
-    # Choose your license
-    license='Apache License 2.0',
-
-    # See https://pypi.python.org/pypi?%3Aaction=list_classifiers
-    classifiers=[
-        'Development Status :: 5 - Production/Stable',
-
-        'Intended Audience :: Developers',
-
-        # that you indicate whether you support Python 2, Python 3 or both.
-        'Programming Language :: Python :: 3.7',
-        'Programming Language :: Python :: 3.8',
-        'Programming Language :: Python :: 3.9',
-        'Programming Language :: Python :: 3.10',
-        'Topic :: Scientific/Engineering :: Visualization',
-        'Topic :: Software Development :: Libraries :: Python Modules',
-        'Topic :: Database',
-    ],
-
-    # What does your project relate to?
-    keywords='sql relational databases ER diagram render',
-
-    # You can just specify the packages manually here if your project is
-    # simple. Or you can use find_packages().
-    packages=[
-        'eralchemy2',
-    ],
-
-    # List run-time dependencies here.  These will be installed by pip when
-    # your project is installed. For an analysis of "install_requires" vs pip's
-    # requirements files see:
-    # https://packaging.python.org/en/latest/requirements.html
-    install_requires=[
-        'SQLAlchemy',
-        'pygraphviz'
-    ],
-    entry_points={
-        'console_scripts': [
-            'eralchemy2=eralchemy2.main:cli',
+    assert "version" in locals()
+    return locals()["version"]
+
+
+if __name__ == "__main__":
+    setup(
+        name="eralchemy2",
+        version=read_version(),
+        description="Simple entity relation (ER) diagrams generation",
+        long_description=read_long_description(),
+        url="https://github.com/maurerle/eralchemy2",
+        author="Florian Maurer",
+        author_email="fmaurer+github@disroot.org",
+        license="Apache License 2.0",
+        classifiers=[
+            "Development Status :: 5 - Production/Stable",
+            "Intended Audience :: Developers",
+            "Programming Language :: Python :: 3.7",
+            "Programming Language :: Python :: 3.8",
+            "Programming Language :: Python :: 3.9",
+            "Programming Language :: Python :: 3.10",
+            "Topic :: Scientific/Engineering :: Visualization",
+            "Topic :: Software Development :: Libraries :: Python Modules",
+            "Topic :: Database",
         ],
-    },
-)
+        keywords="sql relational databases ER diagram render",
+        packages=["eralchemy2"],
+        extras_require={"dev": ["black==22.8.0", "isort==5.10.1"]},
```

Comment:
is there a specific reason to have this version fixed?
If people want to use other packages which rely on a different version of black or isort next to it they will have a hard time figuring out the dependencies

Reply:
Do you expect that people install this package with dev dependencies?

IMHO, this normally only happens if you work in the project and the you are in a separated environment.

Also, if you add a CI at one point and black/isort updates, then the CI might fail out of the sudden

Reply:
Having a CI fail on a future version is an indicator that some feature we are using will be deprecated in the future and we need a change in the future anyway. So this is more like a feature than a big in my point of view.
Having tests run in CI with the latest version is a feature to see if it still works as I don't update my environment everytime.

In my opinion it's better to only have fixed versions if this is required due to incompatibilities, and I don't want to change the extras_require every time just to update the version.

Those discussions are getting quite philosophically fast, and I said that especially for dev-reqs I don't think that they have to be fixed, while you are saying that especially for dev-reqs everyone would have a separate env anyway.

I don't know if there is a pythonic consensus on how to handle this, but I know that spyder and flask do it that way and did not find a major project having fixed version dependencies.

Reply:
I removed the pins